lib/install.js

```diff
@@ -216,9 +217,46 @@ function Installer (where, dryrun, args) {
 }
 Installer.prototype = {}
 
-Installer.prototype.run = function (cb) {
+function updateStatsFile (itWorked) {
+  // If the stats reporter hasn't managed to PUT yet then kill it so that it doesn't
+  // step on our updating the cli-stats json
+  if (npm.statsProcess) npm.statsProcess.kill('SIGKILL')
+  var statsFile = path.join(npm.config.get('cache'), 'anonymous-cli-stats.json')
+  var stats
+  try {
+    stats = JSON.parse(fs.readFileSync(statsFile))
```

Comment:
I've just pushed a little refactor that moves all of the existing logic into a single file (without changing any of those interfaces), which makes this a bit clearer.

At the moment the entire interface that npm has is `saveMetrics` with `true` or `false`. And yeah, that's clearly not going to be sufficient going forward.

On the other hand, I think I'm inclined to leave it as is until we add another metric. My inclination is to wait until we have a concrete addition that necessitates the change in interface. Right now it'd be pretty hypothetical. We can make educated guesses, sure, but front loading the work uh... hmm, doesn't right?